### ISSUE TITLE
Support `ISTIO_META_POD_PORTS` in `ServiceEntry` service registry controller

### ISSUE THREAD

**Issue by wtzhang23** [istio/istio#49926]
**Describe the feature request**
I would like to integrate multitenant, dockerized VM workloads to the mesh. One feature besides #49821 that we'd need is the ability to specify container ports (the port the service binds to) that are different from the `WorkloadEntry.ports` field. I see that the Kubernetes controller searches `proxy.metadata.PodPorts` to register these ports and would like to know if that can be exported to the service entry controller.

The reason why I'd like this is to automatically generate inbound listeners without relying on `Sidecar` definitions.

**Describe alternatives you've considered**
* Use new environment variable to avoid breaking users that accidentally specified this metadata

**Affected product area (please put an X in all that apply)**

[ ] Ambient
[ ] Docs
[ ] Dual Stack
[ ] Installation
[x] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Affected features (please put an X in all that apply)**

[ ] Multi Cluster
[x] Virtual Machine
[ ] Multi Control Plane

**Additional context**
* TLDR is I'd like to support registering Nomad allocations using [bridge networking](https://developer.hashicorp.com/nomad/docs/networking#bridge-networking) by discovering and registering them over MCP + Consul

**Comment by wtzhang23** [comment]
I'm open to working on both issues as long as they are a good idea. It would be my first contribution though...

**Comment by howardjohn** [comment]
Can you give an example of how you would configure this?Its not entirely clear to me

**Comment by wtzhang23** [comment]
I use the `proxyv2` image instead of the `istio-sidecar.deb` as recommended by the docs. While deploying the `proxyv2` image, I have control over the environmental variables and the proxy config to bootstrap pilot agent. To configure this, I'd "manually" (I can do it in an automated way similar to sidecar injection) have to define the environmental variable.

**Comment by wtzhang23** [comment]
E.g. A stripped down Nomad job that I use:
```
job "testbox" {
…


  group "testbox" { # shares cgroup

    network {
	# can use these to infer ISTIO_META_POD_PORTS
      port "http" {
        to = 8080
      }
      port "grpc" {
        to = 8081
      }
      mode = "bridge"
    }

    task “testbox {
      driver         = "docker"
      service { # can use these to infer service entry (see Consul MCP)
        name = "testbox"
	   port = “http”
      }

      service {
        name = "testbox-grpc"
        port = "grpc"
      }
    }
    task "testbox-istio-init" {
      driver = "docker"
      lifecycle {
        hook    = "prestart"
        sidecar = false
      }
      config {
        image   = "istio/proxyv2:1.19.3"
        cap_add = ["net_admin", "net_raw"]
        args    = ["istio-iptables", "-p", "15001", "-z", "15006", "-u", "1337", "-g", "1337", "-m", "REDIRECT", "-i", "*", "-x", "", "-b", "*", "-d", "15090,15021,15020"]
      }
      env {
        ISTIO_META_DNS_CAPTURE       = "true"
        ISTIO_META_DNS_AUTO_ALLOCATE = "true"
      }
    }
    task "testbox-istio-proxy" {
      driver = "docker"
      lifecycle {
        hook    = "prestart"
        sidecar = true
      }
      user = "1337:1337"
      resources {
        cpu    = 4096
        memory = 2048
      }
      config {
        image = "istio/proxyv2:1.19.3"
        args  = ["proxy", "sidecar"]
        mount {
          type     = "bind"
          source   = "secrets"
          target   = "/var/run/secrets"
          readonly = false
        }
        mount {
          type     = "bind"
          source   = "local"
          target   = "/etc/istio/proxy"
          readonly = false
        }
      }
      template {
        data        = <<EOH
…
EOH
        change_mode = "restart"
        destination = "secrets/istio/root-cert.pem"
      }
      template {
        data        = <<EOH
…
EOH
        change_mode = "noop"
        destination = "secrets/tokens/istio-token"
      }
      env {
        // environment variables go here
	   CA_ADDR = …
      }
    }
  }
}
```

**Comment by howardjohn** [comment]
I more meant why do you need to set pod_ports? Like why does `WorkloadEntry.ports` not work?

**Comment by wtzhang23** [comment]
The host port where the workload is exposed externally is different from the port the service is listening to (the allocator randomly provisions a host port to map to the container port). As such, to make the service routable by other workloads, I define the workload entries with `address = <node address>` and `port = <host port>`. However, this leads to istiod to build inbound clusters for the host port instead of the port the service inside the container is listening to. This is a problem because the port is translated from the host port to the service port when the request enters the container.

A sample dump from my mcp server
```
{
    "@type": "type.googleapis.com/istio.mcp.v1alpha1.Resource",
    "metadata": {
        "name": "bar/foo",
        "createTime": "2024-03-14T18:11:24.407105331Z",
        "version": "2024-03-14T18:11:24.407004621Z"
    },
    "body": {
        "@type": "type.googleapis.com/istio.networking.v1alpha3.ServiceEntry",
        "hosts": [
            "foo.bar"
        ],
        "ports": [
            {
                "number": 80,
                "protocol": "http2",
                "name": "consul-service-port"
            }
        ],
        "resolution": "STATIC",
        "endpoints": [
            {
                "address": "<host ip>",
                "ports": {
                    "consul-service-port": <host port>
                }
            }
        ],
        "exportTo": [
            "*"
        ]
    }
}
```

**Comment by howardjohn** [comment]
Ah ok thanks ,I get it now. Yeah I think this is a unique case compare to k8s. In k8s there are 2 ports (service and container), here we have 3 (service, VM outer port, VM inner port). 

Currently the only way to express that is with Sidecar `ingress`. In theory we could add some env var like you said

**Comment by wtzhang23** [comment]
If y'all think this is a good idea, I'm open to give an attempt in implementing a PR. Otherwise, I'll either generate `Sidecar` resources or maintain a private fork for my special use case. Not super urgent to decide soon since I'm in the early stages where the inbound passthrough cluster seems to be sufficient for now,

**Comment by howardjohn** [comment]
Imo Sidecar resource is probably the best bet for now, I think it was actually made to solve this specific case (and others)

**Comment by wtzhang23** [comment]
`Sidecar` will only be feasible if I find a fix for the [issue](https://github.com/istio/istio/issues/49821) linked in the description or just keep WorkloadEntry.labels empty and solely rely on `ISTIO_METAJSON_LABELS` (to handle the edge case of multiple workloads of the same namespace being deployed on the same node and istiod selecting another workload's labels in `GetProxyWorkloadLabels`).

**Comment by wtzhang23** [comment]
Coming back to this issue. Do you know if the `Sidecar` resource is sufficient to handle the case where `H2UpgradePolicy` is true within a DestinationRule but the target service is HTTP 1.x?
